pkg/kube/client_factory.go

```diff
@@ -70,22 +71,30 @@ func newClientFactory(clientConfig clientcmd.ClientConfig, diskCache bool) util.
 			discoveryCacheDir := computeDiscoverCacheDir(filepath.Join(cacheDir, "discovery"), restConfig.Host)
 
 			return diskcached.NewCachedDiscoveryClientForConfig(restConfig, discoveryCacheDir, httpCacheDir, 6*time.Hour)
```

Comment:
@howardjohn how do we know here that the cached has expired? is 6 hours hard coded expiration time? Thanks.

Reply:
6hrs is the value used by kubectl. There is some automatic logic to evict the cache on miss, its not just a static 6 hours.

The logic here makes istioctl identical to kubectl. It also makes it identical to the recommended way to write CLI tools (using https://pkg.go.dev/k8s.io/cli-runtime/pkg/genericclioptions)